drivers/sensor/lis2dux12/lis2dux12.c

```diff
@@ -0,0 +1,412 @@
+
+	return 0;
+}
+
+#ifdef CONFIG_LIS2DUX12_ENABLE_TEMP
+static int lis2dux12_sample_fetch_temp(const struct device *dev)
+{
+	struct lis2dux12_data *data = dev->data;
+	const struct lis2dux12_config *cfg = dev->config;
+	stmdev_ctx_t *ctx = (stmdev_ctx_t *)&cfg->ctx;
+
+	/* fetch raw data sample */
+	lis2dux12_md_t mode;
+	lis2dux12_outt_data_t temp_data = {0};
+
+	if (lis2dux12_outt_data_get(ctx, &mode, &temp_data) < 0) {
+		LOG_ERR("Failed to fetch raw temperature data sample");
+		return -EIO;
+	}
+
+	data->sample_temp = sys_le16_to_cpu(temp_data.heat.raw);
+
+	return 0;
+}
+#endif
+
+static int lis2dux12_sample_fetch(const struct device *dev, enum sensor_channel chan)
+{
+	switch (chan) {
+	case SENSOR_CHAN_ACCEL_XYZ:
+		lis2dux12_sample_fetch_accel(dev);
+		break;
+#if defined(CONFIG_LIS2DUX12_ENABLE_TEMP)
+	case SENSOR_CHAN_DIE_TEMP:
+		lis2dux12_sample_fetch_temp(dev);
+		break;
+#endif
+	case SENSOR_CHAN_ALL:
+		lis2dux12_sample_fetch_accel(dev);
+#if defined(CONFIG_LIS2DUX12_ENABLE_TEMP)
+		lis2dux12_sample_fetch_temp(dev);
+#endif
+		break;
+	default:
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static inline void lis2dux12_convert(struct sensor_value *val, int raw_val, float gain)
+{
+	int64_t dval;
+
+	/* Gain is in mg/LSB */
+	/* Convert to m/s^2 */
+	dval = ((int64_t)raw_val * gain * SENSOR_G) / 1000;
+	val->val1 = dval / 1000000LL;
+	val->val2 = dval % 1000000LL;
+}
+
+static inline int lis2dux12_get_channel(enum sensor_channel chan, struct sensor_value *val,
+					struct lis2dux12_data *data, float gain)
+{
+	switch (chan) {
+	case SENSOR_CHAN_ACCEL_X:
+		lis2dux12_convert(val, data->sample_x, gain);
+		break;
+	case SENSOR_CHAN_ACCEL_Y:
+		lis2dux12_convert(val, data->sample_y, gain);
+		break;
+	case SENSOR_CHAN_ACCEL_Z:
+		lis2dux12_convert(val, data->sample_z, gain);
+		break;
+	case SENSOR_CHAN_ACCEL_XYZ:
+		lis2dux12_convert(val, data->sample_x, gain);
+		lis2dux12_convert(val + 1, data->sample_y, gain);
+		lis2dux12_convert(val + 2, data->sample_z, gain);
+		break;
+#if defined(CONFIG_LIS2DUX12_ENABLE_TEMP)
+	case SENSOR_CHAN_DIE_TEMP:
+		sensor_value_from_double(val, data->sample_temp);
+		break;
+#endif
+	default:
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static int lis2dux12_channel_get(const struct device *dev, enum sensor_channel chan,
+				 struct sensor_value *val)
+{
+	struct lis2dux12_data *data = dev->data;
+
+	return lis2dux12_get_channel(chan, val, data, data->gain);
+}
+
+static const struct sensor_driver_api lis2dux12_driver_api = {
+	.attr_set = lis2dux12_attr_set,
+#if defined(CONFIG_LIS2DUX12_TRIGGER)
+	.trigger_set = lis2dux12_trigger_set,
+#endif
+	.sample_fetch = lis2dux12_sample_fetch,
+	.channel_get = lis2dux12_channel_get,
+};
+
+static int lis2dux12_init(const struct device *dev)
+{
+	const struct lis2dux12_config *const cfg = dev->config;
+	stmdev_ctx_t *ctx = (stmdev_ctx_t *)&cfg->ctx;
+	uint8_t chip_id;
+	int ret;
+
+	lis2dux12_exit_deep_power_down(ctx);
+	k_busy_wait(25000); /* 25ms */
```

Comment:
if you feel like you need to add a comment on a line of code, you probably need either a define or an intermediate variable instead :-)

add a define for this, something like `#define XXX_INIT_DELAY_MS 25` and then do `k_busy_wait(XXX_INIT_DELAY_MS * USEC_PER_MSEC);`.